### ISSUE TITLE
Tuples: inference for method type args fails when tuple names are different

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#14247]
``` C# 
        [WorkItem(10800, "https://github.com/dotnet/roslyn/issues/10800")]
        [Fact]
        public void Inference11()
        {
            var source = @"
class C
{
    static void Main()
    {
        // byref tuple, as a whole, sets an exact bound, names are honored (just like in the case of dynamic
        var ab = (a: 1, b: 2);
        var cd = (c: 1, d: 2);

        // this is an error  since we have two exact bounds
        Test3(ref ab, ref cd);

        // these are ok, since one of the bounds is a lower bound
        Test1(ref ab, cd);
        Test2(ab, ref cd);
}

    static void Test1<T>(ref T x, T y)
    {
        System.Console.WriteLine(typeof(T));
    }

    static void Test2<T>(T x, ref T y)
    {
        System.Console.WriteLine(typeof(T));
    }

    static void Test3<T>(ref T x, ref T y)
    {
        System.Console.WriteLine(typeof(T));
    }
}
" + trivial2uple;

            var comp = CreateCompilationWithMscorlib(source);
            comp.VerifyDiagnostics(
                );
        }
```

Currently, there are unexpected diagnostics.

``` C#
   // (11,9): error CS0411: The type arguments for method 'C.Test3<T>(ref T, ref T)' cannot be inferred from the usage. Try specifying the type arguments explicitly.
                //         Test3(ref ab, ref cd);
                Diagnostic(ErrorCode.ERR_CantInferMethTypeArgs, "Test3").WithArguments("C.Test3<T>(ref T, ref T)").WithLocation(11, 9)
```

**Comment by jcouv** [comment]
Fixed by #16115
